src/VisualStudio/Core/Def/Options/PackageSettingsPersister.cs

```diff
@@ -3,54 +3,27 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Threading;
-using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
 using Microsoft.CodeAnalysis.Options;
 using Microsoft.CodeAnalysis.SolutionCrawler;
 using Microsoft.VisualStudio.LanguageServices.Setup;
-using Microsoft.VisualStudio.Shell;
-using Microsoft.VisualStudio.Threading;
-using Roslyn.Utilities;
 
 namespace Microsoft.VisualStudio.LanguageServices.Options
 {
-    internal sealed class PackageSettingsPersister : IOptionPersister
+    internal sealed class PackageSettingsPersister(IGlobalOptionService optionService) : IOptionPersister
     {
-        private readonly IThreadingContext _threadingContext;
-        private readonly IAsyncServiceProvider _serviceProvider;
-        private readonly IGlobalOptionService _optionService;
         private RoslynPackage? _lazyRoslynPackage;
 
-        public PackageSettingsPersister(
-            IThreadingContext threadingContext,
-            IAsyncServiceProvider serviceProvider,
-            IGlobalOptionService optionService)
+        public void Initialize(RoslynPackage package)
         {
-            _threadingContext = threadingContext;
-            _serviceProvider = serviceProvider;
-            _optionService = optionService;
-
-            // Start the process of loading the Roslyn package and getting options, but don't wait for it to complete.
-            // The setting will be refreshed once available.
-            InitializeAsync(_threadingContext.DisposalToken).ReportNonFatalErrorAsync().Forget();
```

Comment:
My concern here is nothing is ensuring the package might actually load at all -- if you open a CPS-only solution and no documents are open, I don't imagine we'd get a package load at all. And even if you run that case under a debugger and it works _now_, that might only be because of somebody else randomly triggering the load which means we're relying on side effects of somebody other component in VS.

Reply:
I think it might be best to keep the new pattern as is (that the package load initializes this), but still have the creation of the persister still trigger a load of the package. But unlike before we can just call LoadPackageAsync() and not worry about getting anything back -- it's just a fire and forget to make sure that process is started.